### PR TITLE
Vibhansa/pathmessup

### PR DESCRIPTION
**Pull request by vibhansa-msft**
When blob name is merged with temp cache path to form a local path, in some combination its reaching out of temp-cache directory path. Blobfuse shall invalidate such paths and return ENOENT

### CODE REVIEW

**Review by gapra-msft — APPROVED** [review]
Some of these PRs have remnants of other PRs. But looks good to me.

File: blobfuse/utilities.cpp [hunk 1]
```diff
@@ -183,6 +184,20 @@ int azs_getattr(const char *path, struct stat *stbuf)
             }
             else
             {
+                char real_path[PATH_MAX];
+                char *result = realpath(mntPathString.c_str(), real_path);
+                if (result != NULL) {
+                    // Path got resolved. lets check we are not moving away from temp path
+                    // If the file name was created using \..\..\..\abc.txt kind of pattern then mergin it with temp cache path
```
Comment (tasherif-msft) [inline blobfuse/utilities.cpp:191]:
```suggestion
                    // If the file name was created using \..\..\..\abc.txt kind of pattern then merging it with temp cache path
```

File: blobfuse/utilities.cpp [hunk 2]
```diff
@@ -183,6 +184,20 @@ int azs_getattr(const char *path, struct stat *stbuf)
...
+                    // may lead the path out of the temp cache folde rand blobfuse may end up deleting those file if open is hit
```
Comment (tasherif-msft) [inline blobfuse/utilities.cpp:192]:
```suggestion
                    // may lead the path out of the temp cache folder and blobfuse may end up deleting those files if open is hit
```

File: blobfuse/utilities.cpp [hunk 3]
```diff
@@ -183,6 +184,20 @@ int azs_getattr(const char *path, struct stat *stbuf)
...
+                    // To restrict that here we stop getattr and return enoent so that open is never called for such path
+                    // any absolute path going out of temp cache path needs to be invalidated
+                    std::string realPathString(real_path);
+                    if (std::string::npos == realPathString.find(config_options.absoluateTmpPath)) {
```
Comment (tasherif-msft) [inline blobfuse/utilities.cpp:196]: Let's have a proper PR description/context for this fix please!

File: blobfuse/utilities.cpp (hunk 1 above)
Comment (vibhansa-msft) [reply blobfuse/utilities.cpp:191]: Corrected

File: blobfuse/utilities.cpp (hunk 2 above)
Comment (vibhansa-msft) [reply blobfuse/utilities.cpp]: corrected

**Pull request merged**